core/UrlHelper.php

```diff
@@ -155,6 +155,14 @@ public static function getArrayFromQueryString($urlQuery)
         if (strlen($urlQuery) == 0) {
             return array();
         }
+
```

Comment:
I think in the long term adding a cache to UrlHelper.php is a bad idea; we should add the cache at a higher level. However, the tracker isn't going to be coded well enough to allow this anytime soon. If you agree the caching should ideally be somewhere else, can we add a TODO to move it eventually? Or a new issue would work.